sklearn/ensemble/_bagging.py

```diff
@@ -835,7 +841,9 @@ def __init__(
 
     def _get_estimator(self):
         """Resolve which estimator to return (default is DecisionTreeClassifier)"""
-        return self.estimator or DecisionTreeClassifier()
+        if self.estimator is None:
+            return DecisionTreeClassifier()
+        return self.estimator
```

Comment:
why the diff here? I find the existing code clear and shorter

Reply:
Because it fails when enabling metadata routing with a non-default estimator.
See this https://github.com/scikit-learn/scikit-learn/pull/28734#issuecomment-2029446912